src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2808,6 +2808,10 @@ protected function castAttribute($key, $value)
                 return json_decode($value, true);
             case 'collection':
                 return new BaseCollection(json_decode($value, true));
+            case 'date':
+                return Carbon::createFromFormat('Y-m-d', $value);
+            case 'datetime':
+                return Carbon::createFromFormat('Y-m-d H:i:s', $value);
```

Comment:
Need to get the value from format, since different database grammar would have different formatting.

Or why not use the available methods to convert it to carbon object (if possible).